knative-operator/pkg/common/monitoring.go

```diff
@@ -36,6 +42,36 @@ func SetupMonitoringRequirements(api client.Client, instance mf.Owner) error {
 	return nil
 }
 
+func RemoveOldServiceMonitorResources(namespace string, api client.Client) error {
+	currentSMName := os.Getenv(operatorServiceMonitorNameEnvKey)
+	if currentSMName == "" {
+		currentSMName = operatorServiceMonitorDefaultName
+	}
```

Comment:
Is the defaulting necessary? Seems like we use explicit values everywhere, so I'd rather drop this to catch accidental issues.

Reply:
I could remove it yes.